File: Termometro.cs
using System;

class Termometro { 
    private float temperatura; 

    public Termometro() { 
        Random rnd = new Random();
        this.temperatura = (float)(rnd.NextDouble() * 100.0);
    }

    public Termometro(float temperatura) { 
        this.temperatura = temperatura;
    }

    public float TemperaturaC() {  
        return this.temperatura;
    }

    public float TemperaturaF() { 
        return (this.temperatura * 9.0f / 5.0f) + 32.0f;
    }

    public float Temperaturak() { 
        return this.temperatura + 273.15f;
    }

    public void mostrarTemperaturas() { // 
        Console.WriteLine($"Temperatura en Celsius: {TemperaturaC():F2} °C");
        Console.WriteLine($"Temperatura en Fahrenheit: {TemperaturaF():F2} °F");
        Console.WriteLine($"Temperatura en Kelvin: {Temperaturak():F2} K");
    }
}

class Test { 
    static void Main(string[] args) { 
        Console.WriteLine("=== Instancia 1: Constructor por Defecto ===");
        Termometro t1 = new Termometro(); 
        t1.mostrarTemperaturas();

        Console.WriteLine("\n=== Instancia 2: Constructor Parametrizado ===");
        Console.Write("Ingresa la temperatura en grados Celsius: "); 
        
        if (float.TryParse(Console.ReadLine(), out float inputTemp)) {
            Termometro t2 = new Termometro(inputTemp); 
            t2.mostrarTemperaturas();
        } else {
            Console.WriteLine("Entrada no válida. Debes ingresar un número flotante.");
        }
    }
}

File: Termometro.py
import random

class Termometro:
    def __init__(self, temperatura=None):
        if temperatura is None:
            self.__temperatura = random.uniform(0.0, 100.0)
        else:
            # Constructor parametrizado
            self.__temperatura = float(temperatura)

    def TemperaturaC(self):
        return self.__temperatura

    def TemperaturaF(self):
        return (self.__temperatura * 9 / 5) + 32

    def Temperaturak(self):
        return self.__temperatura + 273.15

    def mostrarTemperaturas(self):
        print(f"Temperatura en Celsius: {self.TemperaturaC():.2f} °C")
        print(f"Temperatura en Fahrenheit: {self.TemperaturaF():.2f} °F")
        print(f"Temperatura en Kelvin: {self.Temperaturak():.2f} K")

def main():
    print("=== Instancia 1: Constructor por Defecto ===")
    t1 = Termometro() 
    t1.mostrarTemperaturas()
    
    print("\n=== Instancia 2: Constructor Parametrizado ===")
    try:
        input_temp = float(input("Ingresa la temperatura en grados Celsius: "))
        t2 = Termometro(input_temp) 
        t2.mostrarTemperaturas()
    except ValueError:
        print("Error: Debes ingresar un valor numérico.")

if __name__ == "__main__":
    main()